core/security.js

```diff
@@ -0,0 +1,37 @@
+class Security {
+  constructor() {
+    this.supportsTrustedTypes =
+      window.trustedTypes && window.trustedTypes.createPolicy;
+    let trustedTypesPolicy;
+
+    try {
+      if (this.supportsTrustedTypes) {
+        trustedTypesPolicy = window.trustedTypes.createPolicy('quill@core', {
+          createHTML: function (html) {
+            // Trust all strings passed to this function.
+            return html;
+          },
+        });
+      }
+    } catch (e) {}
+
+    this.blessHTML = trustedTypesPolicy ? trustedTypesPolicy.createHTML.bind(trustedTypesPolicy) : (x) => x;
+  }
+
+  deepBless(input) {
+    return JSON.parse(JSON.stringify(input), (key, value) => {
+      return typeof value === 'string' ? this.blessHTML(value) : value
+    });
+  }
+
+  isTrustedHTML(value) {
+    // eslint-disable-next-line no-undef
+    return this.supportsTrustedTypes && value instanceof TrustedHTML;
```

Comment:
I believe it's possible to spoof the instanceof check. Instead we have `window.trustedTypes.isHTML(value)`, although you'll still have to guard it with the `supportsTrustedTypes` check for backwards compatibility. More on this: https://w3c.github.io/webappsec-trusted-types/dist/spec/#dom-trustedtypepolicyfactory-ishtml